archive/rpm.ksy

```diff
@@ -0,0 +1,1458 @@
+    1133:
+      id: patches_name_obsolete
+      -orig-id: RPMTAG_PATCHESNAME
+    1134:
+      id: patches_flags_obsolete
+      -orig-id: RPMTAG_PATCHESFLAGS
+    1135:
+      id: patches_version_obsolete
+      -orig-id: RPMTAG_PATCHESVERSION
+    1136:
+      id: cache_ctime_internal
+      -orig-id: RPMTAG_CACHECTIME
+    1137:
+      id: cache_pkg_path_internal
+      -orig-id: RPMTAG_CACHEPKGPATH
+    1138:
+      id: cache_pkg_size_internal
+      -orig-id: RPMTAG_CACHEPKGSIZE
+    1139:
+      id: cache_pkg_mtime_internal
+      -orig-id: RPMTAG_CACHEPKGMTIME
+    1140:
+      id: file_colors
+      -orig-id: RPMTAG_FILECOLORS
+    1141:
+      id: file_class
+      -orig-id: RPMTAG_FILECLASS
+    1142:
+      id: class_dict
+      -orig-id: RPMTAG_CLASSDICT
+    1143:
+      id: file_depends_idx
+      -orig-id: RPMTAG_FILEDEPENDSX
+      doc: Index into `::depends_dict` denoting start of this file's dependencies.
+    1144:
+      id: file_depends_num
+      -orig-id: RPMTAG_FILEDEPENDSN
+      doc: Number of file dependencies in `::depends_dict`, starting from `::file_depends_idx`
+    1145:
+      id: depends_dict
+      -orig-id: RPMTAG_DEPENDSDICT
+    1146:
+      id: source_pkgid
+      -orig-id: RPMTAG_SOURCEPKGID
+    1147:
+      id: file_contexts_obsolete
+      -orig-id: RPMTAG_FILECONTEXTS
+      doc-ref: https://github.com/rpm-software-management/rpm/blob/911448f2/doc/manual/tags.md#deprecated--obsolete
+    1148:
+      id: fs_contexts_obsolete
+      -orig-id: RPMTAG_FSCONTEXTS
+      doc-ref: https://github.com/rpm-software-management/rpm/blob/911448f2/doc/manual/tags.md#deprecated--obsolete
+    1149:
+      id: re_contexts_obsolete
+      -orig-id: RPMTAG_RECONTEXTS
+      doc-ref: https://github.com/rpm-software-management/rpm/blob/911448f2/doc/manual/tags.md#deprecated--obsolete
+    1150:
+      id: policies
+      -orig-id: RPMTAG_POLICIES
+    1151:
+      id: pre_trans
+      -orig-id: RPMTAG_PRETRANS
+    1152:
+      id: post_trans
+      -orig-id: RPMTAG_POSTTRANS
+    1153:
+      id: pre_trans_prog
+      -orig-id: RPMTAG_PRETRANSPROG
+    1154:
+      id: post_trans_prog
+      -orig-id: RPMTAG_POSTTRANSPROG
+    1155:
+      id: dist_tag
+      -orig-id: RPMTAG_DISTTAG
+    1156:
+      id: old_suggests_name_obsolete
+      -orig-id: RPMTAG_OLDSUGGESTSNAME
+    1157:
+      id: old_suggests_version_obsolete
+      -orig-id: RPMTAG_OLDSUGGESTSVERSION
+    1158:
+      id: old_suggests_flags_obsolete
+      -orig-id: RPMTAG_OLDSUGGESTSFLAGS
+    1159:
+      id: old_enhances_name_obsolete
+      -orig-id: RPMTAG_OLDENHANCESNAME
+    1160:
+      id: old_enhances_version_obsolete
+      -orig-id: RPMTAG_OLDENHANCESVERSION
+    1161:
+      id: old_enhances_flags_obsolete
+      -orig-id: RPMTAG_OLDENHANCESFLAGS
+    1162:
+      id: priority_unimplemented
+      -orig-id: RPMTAG_PRIORITY
+    1163:
+      id: cvsid_unimplemented
+      -orig-id:
+        - RPMTAG_CVSID
+        - RPMTAG_SVNID
```

Comment:
Interestingly, this is the first (merged) spec where `-orig-id` represented as a list of two elements, which looks strange.

Reply:
@Mingun

> Interestingly, this is the first (merged) spec where `-orig-id` represented as a list of two elements

No, it isn't. Search for regex "`-orig-id:\n`".

```
3 results - 2 files

archive\rpm.ksy:
  1050        id: cvsid_unimplemented
  1051:       -orig-id:
  1052:         - RPMTAG_CVSID
  1053          - RPMTAG_SVNID

image\bmp.ksy:
  206    bitmap_header:
  207:     -orig-id:
  208:       - BITMAPCOREHEADER
  209        - OS21XBITMAPHEADER

  460    rgb_record:
  461:     -orig-id:
  462:       - RGB_TRIPLE
  463        - RGB_QUAD
```

> which looks strange

Is it a problem? How would _you_ represent such a thing?

Reply:
I use multielement `-orig-id`s extensively in my specs.

Reply:
Ok, the first used in the enum value. The other `-orig-id` fields my parser just completely ignore.

> Is it a problem? 

No, I've just noticed that my parser failed to parse that. It is strange in that sense why the same thing is referenced by different names in the original spec?

Reply:
>It is strange in that sense why the same thing is referenced by different names in the original spec?

There may be more than 1 "official" stuff. I.e. a PDF document and an official header file. And a lot of non-official, sometimes no official docs at all, but few reverse-engineered open source impls.